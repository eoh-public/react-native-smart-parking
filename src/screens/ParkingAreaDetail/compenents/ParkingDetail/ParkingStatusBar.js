import React, { memo } from 'react';
import { StyleSheet, View } from 'react-native';
import { t } from 'i18n-js';
import { IconOutline } from '@ant-design/icons-react-native';

import Text from '../../../../commons/Text';
import { Colors } from '../../../../configs';

const ParkingStatusBar = memo(
  ({ status, freeFrom, freeTo, preBook, spot_name }) => {
    const isFull = status === t('full');
    const statusInfo = isFull
      ? 'status_full_info'
      : freeFrom === ''
      ? 'status_always_free'
      : 'status_free_info';

    return (
      <View>
        {!preBook && !spot_name ? (
          <View style={[styles.container, styles.statusContainerNotPreBooking]}>
            <View style={styles.statusBar}>
              <IconOutline
                name="info-circle"
                size={20}
                color={Colors.Gray8}
                style={styles.paddingRight16}
              />
              <Text style={styles.statusNotPreBooking}>
                {t('not_support_pre_booking')}
              </Text>
            </View>
          </View>
        ) : status === null ? null : (
          <View
            style={[
              styles.container,
              isFull ? styles.statusContainerFull : styles.statusContainerFree,
            ]}
          >
            <View style={styles.statusBar}>
              <Text style={isFull ? styles.statusFull : styles.statusFree}>
                {status}
              </Text>
              <Text style={styles.statusInfo}>
                {t(statusInfo, { from: freeFrom, to: freeTo })}
              </Text>
            </View>
          </View>
        )}
      </View>
    );
  }
);

const styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    justifyContent: 'center',
    marginLeft: 16,
    marginRight: 16,
    marginBottom: 16,
  },
  statusContainerFull: {
    backgroundColor: Colors.Red1, // Dust red
  },
  statusContainerFree: {
    backgroundColor: Colors.Green8, // Polar green
  },
  statusContainerNotPreBooking: {
    backgroundColor: Colors.Gray3,
  },
  statusBar: {
    marginLeft: 20,
    flexDirection: 'row',
    alignItems: 'center',
    paddingVertical: 8,
  },
  statusFull: {
    fontWeight: 'bold',
    color: Colors.Red,
  },
  statusFree: {
    fontWeight: 'bold',
    color: Colors.Green,
  },
  statusNotPreBooking: {
    color: Colors.Gray7,
  },
  statusInfo: {
    marginLeft: 10,
    fontSize: 14,
  },
  paddingRight16: {
    paddingRight: 16,
  },
});

export default ParkingStatusBar;
