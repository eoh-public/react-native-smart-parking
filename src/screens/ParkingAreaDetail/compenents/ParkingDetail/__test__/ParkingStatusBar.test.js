import React from 'react';
import { create, act } from 'react-test-renderer';
import moment from 'moment';
import ParkingStatusBar from '../ParkingStatusBar';

describe('Test ParkingStatusBar', () => {
  let tree;
  test('render ParkingStatusBar', () => {
    const obj = { status: 'FULL', freeFrom: '' };
    act(() => {
      tree = create(<ParkingStatusBar {...obj} preBook={true} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('render ParkingStatusBar without preBook', () => {
    const obj = {
      status: '',
      freeFrom: moment('2020-01-01T04:00').utcOffset(0).format('LT'),
      freeTo: moment('2020-01-01T05:00').utcOffset(0).format('LT'),
    };
    act(() => {
      tree = create(<ParkingStatusBar {...obj} preBook={false} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('render ParkingStatusBar without freeFrom', () => {
    const obj = {
      status: '',
      freeFrom: '',
      freeTo: '',
    };
    act(() => {
      tree = create(<ParkingStatusBar {...obj} preBook={true} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('render ParkingStatusBar with status null', () => {
    const obj = {
      status: null,
      freeFrom: '',
      freeTo: '',
    };
    act(() => {
      tree = create(<ParkingStatusBar {...obj} preBook={true} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
