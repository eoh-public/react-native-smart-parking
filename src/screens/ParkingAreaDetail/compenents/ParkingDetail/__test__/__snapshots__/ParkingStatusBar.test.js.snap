// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ParkingStatusBar render ParkingStatusBar 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
          "marginBottom": 16,
          "marginLeft": 16,
          "marginRight": 16,
        },
        Object {
          "backgroundColor": "#FFF1F0",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginLeft": 20,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "color": "red",
              "fontWeight": null,
            },
          ]
        }
      >
        ĐẦY
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "fontSize": 14,
              "marginLeft": 10,
            },
          ]
        }
      >
        Bãi xe hiện tại không có chỗ trống.
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test ParkingStatusBar render ParkingStatusBar with status null 1`] = `<View />`;

exports[`Test ParkingStatusBar render ParkingStatusBar without freeFrom 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
          "marginBottom": 16,
          "marginLeft": 16,
          "marginRight": 16,
        },
        Object {
          "backgroundColor": "#D9F7BE",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginLeft": 20,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "color": "green",
              "fontWeight": null,
            },
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "fontSize": 14,
              "marginLeft": 10,
            },
          ]
        }
      >
        Bãi đỗ xe luôn miễn phí
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test ParkingStatusBar render ParkingStatusBar without preBook 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
          "marginBottom": 16,
          "marginLeft": 16,
          "marginRight": 16,
        },
        Object {
          "backgroundColor": "#F5F5F5",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginLeft": 20,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#595959",
              "fontFamily": "antoutline",
              "fontSize": 20,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "paddingRight": 16,
            },
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "color": "#8C8C8C",
            },
          ]
        }
      >
        Bãi đỗ xe không hỗ trợ đặt chỗ trước.
      </Text>
    </View>
  </View>
</View>
`;
